nexusadspy/client.py

```diff
@@ -124,8 +124,11 @@ def _do_paged_get(self, url, method, params=None, data=None, headers=None,
     def _do_throttled_request(self, url, method, params=None, data=None, headers=None,
                               sec_sleep=2., max_failures=100,
                               get_field=None):
-
-        data = json.dumps(data)
+        try:
+            data = json.dumps(data)
+        except UnicodeDecodeError:
+            # json dump fails for byte stream. Pass data directly in that case.
```

Comment:
Data is gzip file. And segment upload API requires it to be uploaded as POST data.